### DISCUSSION TITLE
How can I group ts-jest with jest?

### DISCUSSION THREAD

**Discussion by oliversalzburg** [renovatebot/renovate#10224] [Request Help] [answered]
I noticed there is an existing `group:jestMonorepo`, so I figured, I could just:
```json
  "packageRules": [
    {
      "matchPackageNames": ["ts-jest"],
      "groupName": "jest monorepo"
    }
  ]
```
And ts-jest would be included in the existing jest monorepo group. But I get 2 MRs. One for jest@27 and one for ts-jest@27.

I also tried:
```json
  "packageRules": [
    {
      "matchPackageNames": ["ts-jest"],
      "matchSourceUrlPrefixes": ["https://github.com/facebook/jest"],
      "groupName": "jest monorepo"
    }
  ]
```
This also doesn't group the MRs together.

The _only_ thing that worked for me so far is to explicitly list all the packages:
```json
"packageRules": [
    {
      "matchPackageNames": ["ts-jest", "jest"],
      "groupName": "jest monorepo"
    }
  ]
```

This will actually group the two upgrades together, but it seemed like the previous approaches (or some variation of them) should also give the desired result.

What is the rigth approach to include related packages like `ts-jest` or `@types/jest` with all of the existing Jest packages?

**Comment by rarkins** [comment, marked as answer]
> But I get 2 MRs. One for jest@27 and one for ts-jest@27.

What are the two branch names?

**Reply by oliversalzburg** [reply]
The branches were named `renovate/jest-27.x` and `renovate/major-jest-monorepo` respectively.

I believe I have since realized part of my mistake though. My configuration consisted purely of the shown rules. I never extended any base configuration. Once I added a `"extends": ["config:js-app"]`, behaviors dramatically changed.

However, one of the behavior changes was that both jest-related MRs got autoclosed for reasons I have yet to understand. I'm assuming it's due to their latest releases being fairly recent.

I am now using this configuration:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:js-app"],
  "packageRules": [
    {
      "matchPackageNames": ["ts-jest"],
      "groupName": "jest monorepo"
    }
  ]
}
```

With my latest renovate run, this resulted in an MR with a branch `renovate/major-jest-monorepo` that will upgrade both ts-jest and jest, as I was asking for. So I hope that was my mistake and what I have now is the solution I was looking for.

**Reply by rarkins** [reply]
The autoclosing likely happened because:
- Your rules change resulted in branch name changes
- Any existing PRs with "wrong" branch then need autoclosing (you can't change the branch name of an in-progress PR)
